keyboards/miiiw/blackio83/rev_0100/rules.mk

```diff
@@ -0,0 +1,29 @@
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = custom
+DIP_SWITCH_ENABLE = yes     # Enable dip switches
+DEBOUNCE_TYPE = sym_defer_pk    # Debounce using eager_pk.
+CUSTOM_MATRIX = lite
+DEFERRED_EXEC_ENABLE = yes
+NO_USB_STARTUP_CHECK = yes
+# EEPROM_DRIVER = i2c
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
+
+OPT_DEFS += -DWS2812
+
+# Project specific files
+SRC += matrix.c \
+	   common/shift_register.c \
+       common/ws2812_custom.c
```

Comment:
```suggestion
# Project specific files
SRC += matrix.c \
	   common/shift_register.c \
```

To add the rgb driver with additional behavior...

Add this:
```make
WS2812_DRIVER_REQUIRED := yes
```
And then properly implement a custom driver.  Eg: 
```c
// LED color buffer
rgb_led_t rgb_matrix_ws2812_array[RGB_MATRIX_LED_COUNT];
bool      ws2812_dirty = false;
static bool p_setup = false;
static bool s_init = false;

static void init(void) {
    palSetLineMode(WS2812_DI_PIN, WS2812_OUTPUT_MODE);
}

void ws2812_poweron(void) {
    if(p_setup) return;
    p_setup = true;
    s_init = false;
    setPinOutput(RGB_EN_PIN);
    writePinHigh(RGB_EN_PIN);
}

void ws2812_poweroff(void) {
    if(!p_setup) return;
    p_setup = false;
    setPinInputLow(WS2812_DI_PIN);
    writePinLow(RGB_EN_PIN);
}

static void flush(void) {
    if (ws2812_dirty && p_setup) {
        ws2812_setleds(rgb_matrix_ws2812_array, RGB_MATRIX_LED_COUNT);
        ws2812_dirty = false;
    }
}

// Set an led in the buffer to a color
static inline void setled(int i, uint8_t r, uint8_t g, uint8_t b) {
#    if defined(RGB_MATRIX_ENABLE) && defined(RGB_MATRIX_SPLIT)
    const uint8_t k_rgb_matrix_split[2] = RGB_MATRIX_SPLIT;
    if (!is_keyboard_left()) {
        if (i >= k_rgb_matrix_split[0]) {
            i -= k_rgb_matrix_split[0];
        } else {
            return;
        }
    } else if (i >= k_rgb_matrix_split[0]) {
        return;
    }
#    endif

    if (rgb_matrix_ws2812_array[i].r == r && rgb_matrix_ws2812_array[i].g == g && rgb_matrix_ws2812_array[i].b == b) {
        return;
    }

    ws2812_dirty                 = true;
    rgb_matrix_ws2812_array[i].r = r;
    rgb_matrix_ws2812_array[i].g = g;
    rgb_matrix_ws2812_array[i].b = b;
#    ifdef RGBW
    convert_rgb_to_rgbw(&rgb_matrix_ws2812_array[i]);
#    endif
}

static void setled_all(uint8_t r, uint8_t g, uint8_t b) {
    for (int i = 0; i < ARRAY_SIZE(rgb_matrix_ws2812_array); i++) {
        setled(i, r, g, b);
    }
}

const rgb_matrix_driver_t rgb_matrix_driver = {
    .init          = init,
    .flush         = flush,
    .set_color     = setled,
    .set_color_all = setled_all,
};
```

This has the advantage of using the core driver files, so it's not a duplicate and out of date and fragmented version of the core file.  (eg, any updates or fixes would be carried over to your board).